### PR TITLE
Light Event Detection Refactor

### PR DESCRIPTION
**Pull request by jarq6c**
This PR cleans up some of the event detection code. It reduces the number of computations and separates the trend computation step from the detrend step more explicitly. This model will make more sense as we add more trend estimation techniques.

## Additions

- `rolling_minimum` function is the method used to estimate the trend.

## Removals

- Function parameters were removed where applicable.

## Changes

- I found a bug in the method that tests event detection on noisy series. The scale and frequency of the noise was incorrect. For some reason I simulated ~1 fluctuation a day, but given the parameters, I clearly meant ~4 fluctuations a day. Also found better results by setting `start_radius` to 1 period over the expected noise frequency (`halflife`). This makes sense since the period of the simulated noise is actually slightly more than `'6H'`.

## Testing

1. Passes all other tests with correction mentioned above.

## Notes

- Nothing in the main entry functions `list_events` and `mark_event_flows` has changed. So nothing downstream should break.

## Todos

- None

## Checklist

- [x] PR has an informative and human-readable title
- [x] PR is well outlined and documented. See [#12](https://github.com/jarq6c/evaluation_tools/pull/12) for an example
- [x] Changes are limited to a single goal (no scope creep)
- [x] Code can be automatically merged (no conflicts)
- [x] Code follows project standards (see [CONTRIBUTING.md](../CONTRIBUTING.md))
- [x] Passes all existing automated tests
- [x] Any _change_ in functionality is tested
- [x] New functions are documented (with a description, list of inputs, and expected output) using [numpy docstring](https://numpydoc.readthedocs.io/en/latest/format.html) formatting
- [ ] Placeholder code is flagged / future todos are captured in comments
- [x] Reviewers requested with the [Reviewers tool](https://help.github.com/articles/requesting-a-pull-request-review/) :arrow_right:

### CODE REVIEW

File: python/events/evaluation_tools/events/event_detection/decomposition.py [hunk 1]
```diff
@@ -22,11 +25,49 @@
...
+        window: int, offset, or BaseIndexer subclass, required
+            Size of the moving window for `pandas.Series.rolling.min`.
+            This filter is used to model the trend in `series`.
+            
+        Returns
+        -------
+        trend: pandas.Series
+            Approximate baseflow trend of original series.
+        
+        """
```
Comment (aaraney) [inline python/events/evaluation_tools/events/event_detection/decomposition.py:49]: It seems appropriate to check if the index of the series is a `pd.DatetimeIndex`. This was done in `detrend_streamflow`. So something like:


```python
if isinstance(series.index, pd.DatetimeIndex) is False:
  # Throw an exception
```

**Review by aaraney — CHANGES_REQUESTED** [review]
Really minor change. Overall looks great!

Interesting your note about the `start_radius` parameter, that makes sense, i'm just curious how we can better estimate the `dt`, the `dt` being how the difference between the `halflife` parameter and the "intended result". Just kind of spitballing, I wonder if `dt` is proportional to the slope change in the `ewma`. Therefore, `dt` would be something like `e^ (something)`. Not really forming a complete idea here, just more or less thinking out loud.

File: python/events/evaluation_tools/events/event_detection/decomposition.py (hunk 1 above)
Comment (aaraney) [reply python/events/evaluation_tools/events/event_detection/decomposition.py:49]: 7af6524 addresses this.

**Review by aaraney — APPROVED** [review]

File: python/events/evaluation_tools/events/event_detection/decomposition.py (hunk 1 above)
Comment (jarq6c) [reply python/events/evaluation_tools/events/event_detection/decomposition.py:49]: Is there a conventional or standard way to check for this?

Examples:
```python
if isinstance(series.index, pd.DatetimeIndex) is False:
  # Throw an exception

if not isinstance(series.index, pd.DatetimeIndex):
  # Throw an exception

if type(series.index) != pd.DatetimeIndex:
  # Throw an exception
```

**Pull request merged**